src/nvim/ex_cmds2.c

```diff
@@ -1847,19 +1852,21 @@ void ex_argdelete(exarg_T *eap)
     /* ":1,4argdel": Delete all arguments in the range. */
     if (eap->line2 > ARGCOUNT)
       eap->line2 = ARGCOUNT;
-    int n = eap->line2 - eap->line1 + 1;
+    long n = eap->line2 - eap->line1 + 1;
     if (*eap->arg != NUL || n <= 0)
       EMSG(_(e_invarg));
     else {
-      for (int i = eap->line1; i <= eap->line2; ++i)
+      assert(eap->line1 <= INT_MAX);
```

Comment:
`w_arg_index` is int anyway, so I decided to cast linenumbers to int. I now have a bad feeling about this. Should we change loop index `i` to long instead, and cast to int only on 1869, since 1868 already guarantees it is safe?